server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/SessionSupplier.scala

```diff
@@ -21,28 +21,24 @@ package org.apache.james.jmap.http
 
 import java.net.URL
 
+import javax.inject.Inject
 import org.apache.james.core.Username
-import org.apache.james.jmap.http.SessionSupplier.HARD_CODED_URL_PREFIX
 import org.apache.james.jmap.model.CapabilityIdentifier.CapabilityIdentifier
 import org.apache.james.jmap.model._
 import reactor.core.scala.publisher.SMono
 
-object SessionSupplier {
-  private val HARD_CODED_URL_PREFIX = "http://this-url-is-hardcoded.org"
-}
-
-class SessionSupplier {
+class SessionSupplier @Inject() (val configuration: JmapRfcConfiguration){
   def generate(username: Username): SMono[Session] = {
     accounts(username)
       .map(account => Session(
         DefaultCapabilities.SUPPORTED,
         List(account),
         primaryAccounts(account.accountId),
         username,
-        apiUrl = new URL(s"$HARD_CODED_URL_PREFIX/jmap"),
-        downloadUrl = new URL(s"$HARD_CODED_URL_PREFIX/download"),
-        uploadUrl = new URL(s"$HARD_CODED_URL_PREFIX/upload"),
-        eventSourceUrl = new URL(s"$HARD_CODED_URL_PREFIX/eventSource")))
+        apiUrl = new URL(s"${configuration.basePath}/jmap"),
+        downloadUrl = new URL(s"${configuration.basePath}/download"),
+        uploadUrl = new URL(s"${configuration.basePath}/upload"),
+        eventSourceUrl = new URL(s"${configuration.basePath}/eventSource")))
```

Comment:
We render a URL as a String in order to get a URL ?

Maybe we can skip the intermediate String representation?